pkg/api/util/ipc_endpoint.go

```diff
@@ -0,0 +1,88 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
+type IPCEndpoint struct {
+	client    *http.Client
+	target    url.URL
+	closeConn bool
+}
+
+func (end *IPCEndpoint) SetCloseConnection(state bool) {
+	end.closeConn = state
+}
+
+// send GET method to the endpoint
+func (end *IPCEndpoint) DoGet() ([]byte, error) {
+	conn := LeaveConnectionOpen
+	if end.closeConn {
+		conn = CloseConnection
+	}
+	// TODO: after removing callers to api/util/DoGet, pass `end.token` instead of using global var
+	res, err := DoGet(end.client, end.target.String(), conn)
+	if err != nil {
+		var errMap = make(map[string]string)
+		_ = json.Unmarshal(res, &errMap) //nolint:errcheck
+		// If the error has been marshalled into a json object, check it and return it properly
+		if errStr, found := errMap["error"]; found {
+			return nil, errors.New(errStr)
+		}
+
+		return nil, fmt.Errorf("Could not reach agent: %v\nMake sure the agent is running before requesting the runtime configuration and contact support if you continue having issues", err)
+	}
+	return res, err
+}
+
+func (end *IPCEndpoint) WithValues(values url.Values) *IPCEndpoint {
+	end.target.RawQuery = values.Encode()
+	return end
+}
+
+func NewIPCEndpoint(config config.Component, endpointPath string) (*IPCEndpoint, error) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Note that this function can only replaces uses with `https` and `cmd_port`, I know some places in the code use `http` (maybe that should be fixed though) and in the core-agent (some will use `cluster_agent.cmd_port`, `process_config.cmd_port`,...)